src/aleph/netty.clj

```diff
@@ -240,19 +240,19 @@
 (defn wrap-future
   [^Future f]
   (when f
-    (if (.isSuccess f)
-      (d/success-deferred (.getNow f) nil)
-      (let [d (d/deferred nil)
-            ;; Ensure the same bindings are installed on the Netty thread (vars,
-            ;; classloader) than the thread registering the
-            ;; `operationComplete` callback.
-            bound-operation-complete (bound-fn* operation-complete)]
-        (.addListener f
-          (reify GenericFutureListener
-            (operationComplete [_ _]
-              (ensure-dynamic-classloader)
-              (bound-operation-complete f d))))
-        d))))
+    (let [d (d/deferred nil)]
+      (if (.isDone f)
+        (operation-complete f d)
```

Comment:
@KingMob in #614 you mentioned:

> For consistency, we can reuse the operation-complete handler, but add a flag to tell it to use/ignore executors (so we can run it on the current thread)

AFAIUI, `(d/deferred nil)` should already take care of this (the `nil` value is the `executor` argument). But please verify :pray:

Reply:
Yeah, I don't know if I had something else in mind, but setting the deferred's executor to nil should work, and replicate the existing behavior in all cases.

I wondered if we should be doing something different for the non-immediate case. By setting the executor to nil, it runs later on a netty thread, where we ran into those issues with the classloader not supporting clojure. If we didn't pass an executor param at all, it'll just default to the current thread's executor, but that's retrieved from the `manifold.executor/executor-thread-local` ThreadLocal, which will be null if we call `wrap-future` from a netty thread.

Ehhh... never mind, the future's listener runs on a netty thread, so even if we set the deferred's executor to `wait-pool` or set the executor ThreadLocal, we'd _still_ have to do the whole classloader dance anyway.

Reply:
I confirm, with `nil` we are sure we will be good.